src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+const exampleDinosaurData = require("../data/dinosaurs");
+const exampleTicketData = require("../data/tickets");
+const exampleRoomData = require("../data/rooms");
+
+/*
+Write a function that will organize all of the dinosaurs by their period key. 
+The function should return an object where the keys are the period values, and 
+the values are an array of IDs.
+*/
+
+/* Expected Result
+  {
+    "Late Jurassic": [
+      "YLtkN9R37",
+      "GGvO1X9Zeh",
+      "BFjjLjea-O",
+      "iOVNUcv-ww",
+      "V53DvdhV2A",
+    ],
+    "Late Cretaceous": [
+      "WHQcpcOj0G",
+      "GKl035EYKN",
+      "2GglUqKT0G",
+      "Pr6kc4Q_Xf",
+      "ft5Gs5izdq",
+      "wuL4ddBinQ",
+    ],
+    // ...
+  }
+*/
+
+function organizeDinosaursByPeriod(dinos, key) {
+  //input an array of objects
+  //output an object where key represents period and value reps an array of dinoIDs
+
+  //create helper function to camelCase
+  function camelCaseKey(keyString) {
+    //create an array that contains all valid keys
+    const validKeys = ["name", "pronunciation", "meaningOfName", "diet", "lengthInMeters", "mya", "info"];
+
+    //iterate through validKeys and if the keyString is included in array then assign it
+    for (let i = 0; i < validKeys.length; i++) {
+      if (validKeys[i].toLowerCase() === keyString.toLowerCase()) {
+        keyString = validKeys[i];
+        break;
+      }
+    }
+
+    return keyString;
+  }
+
+  //if key is undefined assign it the string dinosaurId
+  if (key === undefined) key = "dinosaurId";
+  else key = camelCaseKey(key);
+  //edgecase - all dinos share same keys, so check whether key exists in object
+  if (!dinos[0][key] || key === "period") return "please provide a valid key: (name,pronunciation,meaningOfName,diet,lengthInMeters,mya, or info)";
```

Comment:
While you don't _need_ to use curly braces for `if/else` statements, I would almost always do so. This is pretty confusing to read, especially once the string wraps around to a separate line.

A good case for using single-line `if` statements is when you have some kind of guard clause, like the following:
```js
if (!valid(keys)) return;
```
Other than that, I would stick with curly braces.